### ISSUE TITLE
Laplacian Regularization on GCN 

### ISSUE THREAD

**Issue by sevamoo** [rusty1s/pytorch_geometric#2231]
Hi Thanks a lot for all the materials.  I had a question regarding the Laplacian Regularization (LR) on the loss function for Semi-supervised classification with [GCN paper](https://arxiv.org/pdf/1609.02907.pdf). 
I see that in the paper they skip the regularization part, but I was wondering if you could help with the following questions:

1- What would be the best way to implement the LR using Geometric Pytorch? 
2- In the tensorflow implementation of GCN I see however there is an additional part regarding the regularization, but not sure if I understand it correctly. Here is the one I refer to: [Link](https://github.com/tkipf/gcn/blob/39a4089fe72ad9f055ed6fdb9746abdcfebc4d81/gcn/models.py#L146)

Thanks a lot in advance.

Best
Vahid

**Comment by rusty1s** [comment]
1. This would look similar to:
```python
outputs = model(inputs)
row, col = edge_index
reg_loss = (outputs[row] - outputs[col]).norm(dim=-1).mean()
```
2. This is the weight decay regularization term. We also do this via `torch.optim.Adam(..., weight_decay=...)`

**Comment by sevamoo** [comment]
Thanks a lot! I couldn't imagine this formulation.
